Revision diff navigator

infinite-persistence opened the pull request:
Closes [#5694 Option to step through changes in revision history ](https://github.com/outline/outline/issues/5694)

tommoor commented:
Cool, nice first pass! I would probably consider moving this up until the header area next to the restore button – did you try that?

infinite-persistence commented:
Ah, putting the new component in `Header` was my first choice too, but:
- Would need to put the `editorRef` and `selectedOpIndex` into `uiStore` to communicate with `RevisionViewer`. Is storing ref in store normal?
- Next idea is to just `position: fixed` it beside the Restore button, but that would require listening to `resize` to handle re-adjustments.  Ok to add listeners for this case?
    - If the extra listener is ok, we can also change the highlight-animation to a small overlay marker instead (ala Google Docs). I think it's a better visual indicator, but I dropped it to avoid the extra listener and extra handling for weird zoom ratios.
        - Also considered letting `RevisionViewer` handle a local copy of `revision.html` and change the css (to avoid positioning issues with the overlay method), but I that felt like too much processing.

What do you think?

tommoor commented:
Would not put these in the store, a reference to the editor can be gotten through `useEditor` context. For `selectedOpIndex` you would probably need to move that state in `Document.tsx` and pass through to `RevisionViewer` that way.

infinite-persistence commented:
> a reference to the editor can be gotten through useEditor context

Hm ... for this case, it's not an actual editor, but just a div to display the server-rendered diff?

> For selectedOpIndex you would probably need to move that state in Document.tsx and pass through to RevisionViewer that way.

Ah, cool. I'll take a look.

infinite-persistence commented:
**Update:** Relocate to header, refactor as standalone component.

_See commit messages for details. Deviated from suggestion a bit._
